folium/vector_layers.py

```diff
@@ -159,16 +159,27 @@ class PolyLine(Marker):
                     {{this.location}},
                     {{ this.options }}
                     )
-                    {% if this.tooltip %}.bindTooltip("{{this.tooltip.__str__()}}"){% endif %}
+                    {% if this.tooltip %}.bindTooltip("{{ this.tooltip }}"){% endif %}
```

Comment:
We don't need these lines in all these templates anymore now that we have a Tooltip class! In all cases add a Tooltip object as a child.

Reply:
@Conengmo the reason I had preserved them on these was in the case the user passes a string instead of a `Tooltip` to the `tooltip` arg. This won't break code written for previous folium versions on these classes and lets people just write in simple text for the tooltip, if they don't need our additional `Tooltip` functionality exposed, as I imagine may be the case for looping over many `Marker`'s ,etc.

`self.tooltip = tooltip` only if `isinstance(tooltip, str)`
`self.add_child(tooltip)` only if `isinstance(tooltip, Tooltip)`

and in each object's template, the tooltip is only bound if `self.tooltip` exists:

`{% if self.tooltip  %}.bindTooltip('{{ self.tooltip }}'){% endif %}`

Thoughts?